### DISCUSSION TITLE
Running into problem when using pd.read_sql with SQLAlchemy.

### DISCUSSION THREAD

**Discussion by aimanaziz1993** [oracle/python-oracledb#266] [Q&A]
<img width="824" alt="Screenshot 2023-12-12 at 4 30 55 PM" src="https://github.com/oracle/python-oracledb/assets/43428455/5e91ec99-227b-4091-b3c9-9f6c37732b1b">

Exception:
<oracledb.DbObject MDSYS.SDO_GEOMETRY at 0x1ed8385ce90> _typ

Version:
oracledb==1.4.2

The table I'm trying to read contain MDSYS.SDO_GEOMETRY column which are spatial oracle type.
However, If added the line became like this -> line 42:  attr_impl = self._impl._type.attrs_by_name[name] it works. No exceptions raised.
Is there any workaround to be able to let pandas to read the table and convert them into DataFrame?

Thanks guys for the help!

**Comment by cjbj** [comment]
- please don't include screenshots: not everyone can read them, and we can't cut/paste to test or reply
- please show some code and the SQL to create your tables etc
- should you ask this on the pandas or sqlalchemy projects?

**Comment by anthony-tuininga** [comment]
This has already been [addressed](https://github.com/oracle/python-oracledb/commit/5250faee6f41a7a03c38a2df37edef2ef501292a) and will be released as part of version 2.0. If you are able to [build from source](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code) you can verify that it works for you.
